lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -285,11 +285,16 @@ def development_pod_targets
 
         def add_pod_target_test_dependencies(pod_target, frameworks_group)
           test_dependent_targets = pod_target.all_test_dependent_targets
-          pod_target.test_native_targets.each do |test_native_target|
+          pod_target.test_specs.each do |test_spec|
```

Comment:
here we loop through each test spec instead because we query the test spec if it requires an app host. If it does we wire up the dependency to the app host so the app host gets built.